packages/web/src/pages/GenerateDiagramPage.tsx

```diff
@@ -0,0 +1,697 @@
+  const { pathname, search } = useLocation();
+  const {
+    loading,
+    getModelId,
+    setModelId,
+    clear: clearChat,
+    setLoading,
+    messages,
+    isEmpty,
+    postDiagram,
+    diagramType,
+  } = useDiagram(pathname);
+
+  const { modelIds: availableModels } = MODELS;
+  const modelId = getModelId();
+
+  const disabledExec = useMemo(() => {
+    return content === '' || loading;
+  }, [content, loading]);
+
+  useEffect(() => {
+    (() => {
+      if (search) {
+        const params = queryString.parse(search) as DiagramPageQueryParams;
+        const modelIdFromParams = params.modelId;
+
+        if (params.content) {
+          setContent(params.content);
+        }
+
+        if (modelIdFromParams && availableModels.includes(modelIdFromParams)) {
+          setModelId(modelIdFromParams);
+        } else {
+          setModelId(availableModels[0]);
+        }
+      } else {
+        setModelId(availableModels[0]);
+      }
+    })();
+    // searchのみを依存配列に含める
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [search]);
+
+  // メッセージの処理
+  useEffect(() => {
+    if (messages.length === 0) return;
+
+    const lastMessage = messages[messages.length - 1];
+    if (lastMessage.role !== 'assistant') return;
+
+    const currentMessage = lastMessage.content;
+
+    // Mermaidコードの処理
+    if (currentMessage.toLowerCase().includes('```mermaid')) {
+      const mermaidCode = currentMessage
+        .split('```mermaid')[1]
+        .split('```')[0]
+        .trim();
+      setDiagramCode(mermaidCode);
+    }
+
+    // 説明文の処理
+    if (currentMessage.toLowerCase().includes('<description>')) {
+      const mermaidDescription = currentMessage
+        .split(/<description>/i)[1]
+        .split(/<\/description>/i)[0]
+        .trim();
+      setDiagramSentence(mermaidDescription);
+    } else if (
+      currentMessage.includes(
+        'ただいまアクセスが集中しているため時間をおいて試してみてください。'
+      )
+    ) {
+      setDiagramSentence(
+        'ただいまアクセスが集中しているため時間をおいて試してみてください。'
+      );
+    } else {
+      setDiagramSentence(currentMessage);
+    }
+  }, [messages, setDiagramCode, setDiagramSentence]);
+
+  // ダイアグラム生成処理を実行
+  const onClickExec = useCallback(async () => {
+    if (loading) return;
+    setLoading(true);
+    // content 以外 clear
+    setDiagramGenerationError(null);
+    setDiagramCode('');
+    setDiagramSentence('');
+
+    try {
+      await postDiagram(content, selectedType);
+    } catch (error: unknown) {
+      if (error instanceof Error) setDiagramGenerationError(error);
+      else setDiagramGenerationError(new Error(`${error}`));
+    }
+  }, [
+    content,
+    loading,
+    postDiagram,
+    selectedType,
+    setDiagramCode,
+    setDiagramGenerationError,
+    setDiagramSentence,
+    setLoading,
+  ]);
+
+  // リセット
+  const onClickClear = useCallback(() => {
+    clear();
+    clearChat();
+  }, [clear, clearChat]);
+
+  return (
+    <div className="flex min-h-screen flex-col bg-gray-50 lg:h-screen">
```

Comment:
```suggestion
    <div className="flex min-h-screen flex-col lg:h-screen">
```

他のユースケースも背景は白で統一しているので合わせませんか？
これに依存して回答のブロックや図を描画するブロックなどの背景色はよしなに調整いただけますか？
図を描画する部分はダウンロードすると背景が白くなるので、白でもいいかなと思います。

Reply:
対応しました🫡